client/python/lib/ovmsclient/tfs_compat/grpc/serving_client.py

```diff
@@ -105,7 +116,111 @@ def get_model_status(self, request):
 
     @classmethod
     def _build(cls, config):
-        raise NotImplementedError
+        
+        _check_config(config)
+
+        grpc_address = f"{config['address']}:{config['port']}"
+
+        if 'tls_config' in config:
+            client_key, client_cert, server_cert = _prepare_certs(
+                config['tls_config']['client_key_path'] if 'client_key_path' in config['tls_config'] else None,
+                config['tls_config']['client_cert_path'] if 'client_cert_path' in config['tls_config'] else None,
+                config['tls_config']['server_cert_path']
+            )
+            
+            creds = grpc.ssl_channel_credentials(root_certificates=server_cert,
+            private_key=client_key, certificate_chain=client_cert)
+
+            channel = grpc.secure_channel(grpc_address, creds)
+        else:
+            channel = grpc.insecure_channel(grpc_address)
+
+        prediction_service_stub = prediction_service_pb2_grpc.PredictionServiceStub(channel)
+        model_service_stub = model_service_pb2_grpc.ModelServiceStub(channel)
+        stubs = {
+            "prediction_service_stub" : prediction_service_stub,
+            "model_service_stub" : model_service_stub
+        }
+
+        return cls(channel, stubs)
+
+def _prepare_certs(client_key, client_cert, server_cert):
+    
+    if client_key is not None:
+        with open(client_key, 'rb') as key:
+            client_key = key.read()
+        try:
+            p_key = OpenSSL.crypto.load_privatekey(OpenSSL.crypto.FILETYPE_PEM, client_key)
+        except OpenSSL.crypto.Error as e_info:
+            raise ValueError('client key file is not valid')
+
+    if client_cert is not None:
+        with open(client_cert, 'rb') as ca:
+            client_cert = ca.read()
+        try:
+            c_cert = OpenSSL.crypto.load_certificate(
+                OpenSSL.crypto.FILETYPE_PEM,
+                client_cert
+            )
+        except OpenSSL.crypto.Error as e_info:
+            raise ValueError('client certificate file is not valid')
+        
+    with open(server_cert, 'rb') as f:
+        server_cert = f.read()
+    try:
+        s_cert = OpenSSL.crypto.load_certificate(
+            OpenSSL.crypto.FILETYPE_PEM,
+            server_cert
+        )
+    except OpenSSL.crypto.Error as e_info:
+        raise ValueError('server certificate file is not valid')
```

Comment:
The code is almost the same for both certificates. Please create a method like: `_check_certificate_valid(certificate_path)`
and just use it here to avoid duplication.